### DISCUSSION TITLE
Solution

### DISCUSSION THREAD

**Discussion by WangShiJieAOTM** [foxglove/community#714] [Q&A]
I want to deploy Foxglove on my cloud server, let my robot transmit real-time data to the server, open the deployment studio on any computer that can access the public network, and check the real-time robot data. What should I do?

**Comment by banisadr** [comment]
@WangShiJieAOTM you'll have to be more specific with your questions in order to get quality answers on the discussions forum. Is there a particular part of your architecture or stack you're looking for help with?

**Reply by WangShiJieAOTM** [reply]
I want to deploy foxglove studio on the cloud server so that any device that can access the public network can open the studio, which can display the real-time information of my robot. What should I do to send the real-time information of the robot to the studio on the cloud server?

**Reply by jtbandes** [reply]
One thing you can do is construct a URL that points to a specific data source (e.g. websocket): https://foxglove.dev/docs/studio/building-and-sharing-links#data-sources
Does this help?

**Reply by WangShiJieAOTM** [reply]
My robot does not have a public IP address

**Reply by keogh** [reply]
@WangShiJieAOTM the data source is for the studio so it displays your robot data. 

I'm not very familiar with the platform yet, still based on what I read I think you still need your robot to put its data into the "stream", thus your robot needs to push its data into a websocket endpoint, that websocket then pushes it to studio via the data source connection @jtbandes suggested.

Hopefully it makes sense and please correct me if my assumptions are wrong I don't want to confuse people :P :)

**Reply by WangShiJieAOTM** [reply]
I wrote a topic forwarding program on the bot that subscribed to the bot's own topic and then went to a callback.  In the callback I used roslibpy to send the data to port 9090 on the server.  It seemed to transmit the content successfully, but there was a problem with the frequency, and messages were often stuck or suddenly received quickly.  10HZ point cloud data will appear such a problem, I use 'rosbag play -r 0.25' seems to be continuous, the frequency is normal.  Why is that?  I also tested this solution on the LAN to make sure the bandwidth was adequate

**Reply by WangShiJieAOTM** [reply]
The size of a frame point cloud is 800 kbyte, and 10HZ means 8MByte/s
